minimalkv/_get_store.py

```diff
@@ -32,27 +40,101 @@ def get_store_from_url(url: str) -> KeyValueStore:
-        * AzureBlockBlockStorage (SAS): ``azure://account_name:shared_access_signature@container?use_sas&create_if_missing=false[?max_connections=2&socket_timeout=(20,100)]``
-        * AzureBlockBlockStorage (SAS): ``azure://account_name:shared_access_signature@container?use_sas&create_if_missing=false[?max_connections=2&socket_timeout=(20,100)][?max_block_size=4*1024*1024&max_single_put_size=64*1024*1024]``
-        * GoogleCloudStorage: ``gcs://<base64 encoded credentials JSON>@bucket_name[?create_if_missing=true][&bucket_creation_location=EUROPE-WEST1]``
+        * S3FSStore ``s3://access_key:secret_key@endpoint/bucket[?create_if_missing=true]``
+
+    See the respective store's :func:`_from_parsed_url` function for more details.
+
+    """
+    from minimalkv._hstores import HS3FSStore
+    from minimalkv.net.s3fsstore import S3FSStore
+
+    scheme_to_store: Dict[str, Type[KeyValueStore]] = {
+        "s3": S3FSStore,
+        "hs3": HS3FSStore,
+        "boto": HS3FSStore,
+    }
+
+    parsed_url = urisplit(url)
+    # Wrappers can be used to add functionality to a store, e.g. encryption.
+    # Wrappers are separated by `+` and can be specified in two ways:
+    # 1. As the fragment, e.g. "s3://...#wrap:readonly"
+    # 2. Deprecated: As part of the scheme, e.g. "s3+readonly://..."
+    wrappers = _extract_wrappers(parsed_url)
+
+    # Remove wrappers from scheme
+    scheme_parts = parsed_url.getscheme().split("+")
+    # pop off the type of the store
+    scheme = scheme_parts[0]
+
+    if scheme not in scheme_to_store:
+        # If we can't find the scheme, we fall back to the old creation methods
+        return get_store(**url2dict(url))
+
+    store_cls_from_url = scheme_to_store[scheme]
+    if store_cls is not None and store_cls_from_url != store_cls:
+        raise ValueError(
+            f"URL scheme {scheme} does not match store class {store_cls.__name__}"
+        )
+
+    query_listdict: Dict[str, List[str]] = parsed_url.getquerydict()
+    # We will just use the last occurrence for each key
+    query = {k: v[-1] for k, v in query_listdict.items()}
+
+    store = store_cls_from_url._from_parsed_url(parsed_url, query)
+
+    # apply wrappers/decorators:
+    from minimalkv._store_decoration import decorate_store
+
+    wrapped_store = reduce(decorate_store, wrappers, store)
+
+    return wrapped_store
+
 
-    Get the encoded credentials as string like so:
+def _extract_wrappers(parsed_url: SplitResult) -> List[str]:
+    """
+    Extract wrappers from a parsed URL.
+
+    Wrappers allow you to add additional functionality to a store, e.g. encryption.
+    Wrappers are specified in the fragment part of the URL, e.g. "s3://...#wrap:readonly+urlencode"
 
-    .. code-block:: python
+    Deprecated: wrappers can also be specified as part of the scheme, e.g. "s3+readonly+urlencode://...".
+    This way of specifying wrappers will be removed in a future version.
```

Comment:
Why? I prefer that syntax. It is also a similar approach as I have seen in Hadoop and SQLAlchemy.